src/feeds/DynamicFeeCurveFeed.sol

```diff
@@ -0,0 +1,126 @@
+            pairedTokenToUsd.decimals() == 18,
+            "ChainlinkCurveFeed: DECIMALS_MISMATCH"
+        );
+        curvePool = ICurvePool(_curvePool);
+        uint _index;
+        if(ICurvePool(_curvePool).coins(0) == _asset)
+            _index = 0;
+        else if(ICurvePool(_curvePool).coins(1) == _asset)
+            _index = 1;
+        else
+            revert("CurveFeed: ASSET NOT IN TWO POOL");
+        assetIndex = _index;
+
+        string memory coin = IERC20(_asset).symbol();
+        description = string(abi.encodePacked(coin, " / USD"));
+    }
+
+    event NewMaxFee(int maxFee);
+    event NewGov(address newGov);
+    event NewPendingGov(address newPendingGov);
+
+    /**
+     * @notice Retrieves the latest round data for the pairedToken token price feed
+     * @return roundId The round ID of the Chainlink price feed for the feed with the lowest updatedAt feed
+     * @return usdPrice The latest FiRM asset price in USD
+     * @return startedAt The timestamp when the latest round of Chainlink price feed started of the lowest last updatedAt feed
+     * @return updatedAt The lowest timestamp when either of the latest round of Chainlink price feed was updated
+     * @return answeredInRound The round ID in which the answer was computed of the lowest updatedAt feed
+     */
+    function latestRoundData()
+        public
+        view
+        returns (
+            uint80 roundId,
+            int256 usdPrice,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        )
+    {
+        int256 pairedTokenToUsdPrice;
+        (
+            roundId,
+            pairedTokenToUsdPrice,
+            startedAt,
+            updatedAt,
+            answeredInRound
+        ) = pairedTokenToUsd.latestRoundData();
+        
+        int256 fee = int256(curvePool.fee());
+        if(fee > maxFee) fee = maxFee;
+        //crv oracle price is either asset/pairedToken or pairedToken/asset depending on asset index
+        int256 crvOraclePrice = int256(curvePool.price_oracle());
+        //Depending on assetIndex we either divide or multiply by crv oracle price
+        usdPrice = assetIndex == 0 ?
+            pairedTokenToUsdPrice * 1e18 / crvOraclePrice :
+            crvOraclePrice * pairedTokenToUsdPrice / 1e18;
+        //Reduce the price by the dynamic fee amount
+        usdPrice -= usdPrice * fee / 1e10; 
+    }
+
+    /**
+     * @notice Returns the latest price only
+     * @dev Unlike chainlink oracles, the latestAnswer will always be the same as in the latestRoundData
+     * @return int256 Returns the last finalized price of the chainlink oracle
+     */
+    function latestAnswer() external view returns (int256) {
+        (, int256 latestPrice, , , ) = latestRoundData();
+        return latestPrice;
+    }
+
+    function setMaxFee(int _maxFee) external {
+        require(msg.sender == gov, "ONLY GOV");
+        require(_maxFee <= 1e10, "CurveFeed: maxFee > 100%");
```

Comment:
suggestion: edit require statement to prevent negative values
`require(_maxFee >= 0 && _maxFee <= 1e10)`

if ever a negative value is provided, it will cause this line to always clamp to it, which i presume is not intentional.
https://github.com/InverseFinance/FiRM/blob/c15d14b09251951f5762a57feb4d4c5b14aacfb0/src/feeds/DynamicFeeCurveFeed.sol#L86